src/ui/map.js

```diff
@@ -1216,7 +1216,7 @@ class Map extends Camera {
 
         if (projectionHasChanged) {
             // If a zoom transition on globe
-            if (prevProjection.name === 'globe' && this.getProjection().name === 'globe') {
+            if (this._prevProjection && this._prevProjection.name === 'globe' && this.getProjection().name === 'globe') {
                 this.style._forceSymbolLayerUpdate();
             } else {
                 // If a switch between different projections
```

Comment:
This code path is already a bit complex and I'm worried about adding another state to track (`this._prevProjection`) considering we already have a few bugs with projection switching.

So maybe there's another way to go about fixing that, that may be simpler. If I understand the issue correctly, the displacement issue only happens when a switch between a non-mercator projection and globe projection happens.

If so, would the following work? This is untested code, but I'm curious about whether that would work:

```js
// We hit the threshold globe - mercator transition, only symbol relayout is necessary
const globeHasTransitioned = prevProjection.name === 'globe' && this.getProjection().name === 'globe';
if (globeHasTransitioned) {
    this.style._forceSymbolLayerUpdate();
}
// We switch between a projection with and without dynamic runtime reprojection
// Clearing tiles entirely is necessary
if (prevProjection.isReprojectedInTileSpace && !this.getProjection().isReprojectedInTileSpace) {
    // If a switch between different projections
    this.painter.clearBackgroundTiles();
    for (const id in this.style._sourceCaches) {
        this.style._sourceCaches[id].clearTiles();
    }
}
```

Reply:
@SnailBones also mentioned possibly using a zoom transition to check, so might look into that as well

Reply:
`isReprojectedInTileSpace` did not work, but I changed to using a boolean argument in updateProjection to indicate "forceSymbolLayerUpdate"

Reply:
Hey thanks @karimnaaji - I got it to work using the `isReprojectedInTileSpace`. Using transform.projection allows capturing the prevProjection at the beginning of updateProjection as well.

I do think though we should be clearing tiles switching globe to mercator. I noticed with streets that it doesn't place symbols with this new setup and toggling globe -> mercator until interacting with the map.

Reply:
The name `forceSymbolLayerUpdate` is incomplete, the full effect to choose between either updating symbols (on zoom transition) or clearing tiles (on projection change). IMO a more high level name, e.g. `globeZoomTransition` would better clarify the purpose of this argument.

I agree with the link shared by @karimnaaji that an object instead of argument is worth considering, this codepath shouldn't be called very often so I don't think performance will be an issue. On the other hand, the boolean trap is pretty widespread in our code already.

Reply:
>I do think though we should be clearing tiles (or force symbol placement) switching globe to mercator. I noticed with streets that it doesn't place symbols with this new setup and toggling globe -> mercator until interacting with the map.

>But as @avpeery mentions above I think this is missing the edge case when a developer switches between globe and Mercator projection. In this case, I think we want the same behavior as other projection switches, i.e. clearing tiles and not forcing a symbol layer update.

I think that reloading tiles is already a process already involving symbol relayout since everything gets dropped anyway, so maybe we can simplify it to doing it every time we have a projection change? Like so:

```
if (projectionHasChanged) {
  this.style._forceSymbolLayerUpdate();
  if (this.transform.projection.isReprojectedInTileSpace || prevProjectionReprojected) {
    // reload tiles
  }
}
```

Reply:
@karimnaaji We could flip it. so an "else" trigger the forceSymbolLayerUpdate. Feels redundant to force symbol layer update and then trigger clearing tiles.

Reply:
Working on the render-tests tbd!

Reply:
> I think that reloading tiles is already a process involving symbol update since everything gets dropped anyway, so maybe we can simplify it to doing it every time we have a projection change? Like so:

@karimnaaji I like that this makes it clear that one update is more thorough than the other. But are we confident that the additional calls to `_updateLayer` won't affect performance?

Reply:
@karimnaaji and @SnailBones - I tested out this proposed approach but the call to forceSymbolLayerUpdate broke at least 20 render tests failing to place symbols on globe. I feel like it's too risky of a change and I think being intentional about what the forceSymbolLayerUpdate is supposed to do (i.e. avoiding clearing all the tiles when it's a globe transition) is the better approach

Reply:
@avpeery it might be important to understand why, as this might give us more clue towards simplifying and refactoring this part of the code. 

>But are we confident that the additional calls to _updateLayer won't affect performance?

I can't say for sure whether it affects performance, it could be worth for @avpeery to benchmark the final solution to ensure that once it's finalized. But as this isn't a hot code path and this might be negligible, I'm leaning towards ensuring that we keep the logic as simple as possible.